File: component-acgschedule/src/main/java/com/rabtman/acgschedule/mvp/model/entity/ScheduleTimeItem.java
package com.rabtman.acgschedule.mvp.model.entity;

import com.chad.library.adapter.base.entity.JSectionEntity;
import com.rabtman.acgschedule.mvp.model.jsoup.ScheduleWeek.ScheduleItem;

/**
 * @author Rabtman
 */

public class ScheduleTimeItem extends JSectionEntity {

  public boolean isHeader;
  public String header;
  public int headerIndex;
  public ScheduleItem data;


  public ScheduleTimeItem(boolean isHeader, String header, int index) {
    this.isHeader = isHeader;
    this.header = header;
    this.headerIndex = index;
  }

  public ScheduleTimeItem(ScheduleItem scheduleItem, int index) {
    data = scheduleItem;
    headerIndex = index;
  }

  @Override
  public boolean isHeader() {
    return isHeader;
  }
}

File: component-acgschedule/src/main/java/com/rabtman/acgschedule/mvp/model/entity/ScheduleTimeItem.kt
package com.rabtman.acgschedule.mvp.model.entity

import com.chad.library.adapter.base.entity.JSectionEntity
import com.rabtman.acgschedule.mvp.model.jsoup.ScheduleWeek.ScheduleItem

/**
 * @author Rabtman
 */
class ScheduleTimeItem : JSectionEntity {
    override var isHeader = false

    @JvmField
    var header: String? = null

    @JvmField
    var headerIndex: Int

    @JvmField
    var data: ScheduleItem? = null

    constructor(isHeader: Boolean, header: String?, index: Int) {
        this.isHeader = isHeader
        this.header = header
        headerIndex = index
    }

    constructor(scheduleItem: ScheduleItem?, index: Int) {
        data = scheduleItem
        headerIndex = index
    }

}